src/main/kotlin/com/openlattice/search/SortType.kt

```diff
@@ -0,0 +1,45 @@
+package com.openlattice.search
+
+import com.google.common.collect.ImmutableSet
+import org.apache.olingo.commons.api.edm.EdmPrimitiveTypeKind
+
+private val ALLOWED_FIELD_DATATYPES = ImmutableSet.of(
+        /* dates and times */
+        EdmPrimitiveTypeKind.Date,
+        EdmPrimitiveTypeKind.DateTimeOffset,
+        EdmPrimitiveTypeKind.TimeOfDay,
+
+        /* numbers */
+        EdmPrimitiveTypeKind.Int16,
+        EdmPrimitiveTypeKind.Int32,
+        EdmPrimitiveTypeKind.Int64,
+        EdmPrimitiveTypeKind.Decimal,
+        EdmPrimitiveTypeKind.Double,
+        EdmPrimitiveTypeKind.Duration
+)
+
+private val ALLOWED_GEO_DATATYPES = ImmutableSet.of(
+        EdmPrimitiveTypeKind.GeographyPoint
+)
+
+private val ALLOWED_SCORE_DATATYPES = EdmPrimitiveTypeKind.values().toSet()
+
+enum class SortType {
+    field,
+    score,
+    geoDistance;
+
+    companion object {
+
+        @JvmStatic
+        fun getAllowedDatatypes(sortType: SortType): Set<EdmPrimitiveTypeKind> {
+            return when (sortType) {
+                field -> ALLOWED_FIELD_DATATYPES
+                geoDistance -> ALLOWED_GEO_DATATYPES
+                score -> ALLOWED_SCORE_DATATYPES
+            }
+        }
```

Comment:
This is cool with me the way it is but could also do this:
```
enum class SortType( Set<EdmPrimitiveTypeKind> )
```
Unsure which pattern is preferred

Reply:
yes i like that better